### ISSUE TITLE
the perfomance of face recognition

### ISSUE THREAD

**Issue by Noobzm** [fdbtrs/CR-FIQA#1]
In the provided code,  the loss_qs is directly added to the main CE loss and backward.
Since the label (ccs/nnccs) is no stable, I wonder if this regression task will affect the final performance of backbone/face recognition?  

Thanks in advance!

**Comment by fdbtrs** [comment]
Thank you for the interesting question. I would answer under two points, the first is just to clarify the goal of the CR-FIQA solution, and the second to present a more direct answer to your question:

The presented solution does not aim, and is not presented as, a solution to extract face embeddings, but rather a FIQA solution. This is essential for the FR training to work well so that the CCS/NNCCS values make sense and correspond to a measure of relative classifiablity.
As this is an important point and we also had a similar curiosity that drove your question and motivated in the first point, we did evaluate the CR-FIQA model as an FR embedding extraction model after every training epoch, and on 6 FR benchmarks (LFW, CPLFW, CALFW, Agedp-30, cfp-fp, cfp-ff). We attach here the log file of the training showing this performance at every epoch for the CR-FIQA(L) model. This log file is available in this GitHub repository when accessing pre-trained model (separate file "training.log"). One can notice that the performance is very comparable to the model trained under a similar training setting and only using the FR loss (here ArcFace, reported (https://arxiv.org/abs/1801.07698)  e.g. on lfw ResNet100 trained on ArcFace loss reports an accuracy of 99.83% on LFW, 98.15% on agedb-30, and 92.08% on CPLFW, while the CR-FIQA(L) reaches 99.80% on LFW, 98.20% on Age-db30, and 92.90% on CPLFW).
Thank you again for the interesting question and please do let us know if we can clarify this point further.

**Comment by Noobzm** [comment]
> Thank you for the interesting question. I would answer under two points, the first is just to clarify the goal of the CR-FIQA solution, and the second to present a more direct answer to your question:
> 
> The presented solution does not aim, and is not presented as, a solution to extract face embeddings, but rather a FIQA solution. This is essential for the FR training to work well so that the CCS/NNCCS values make sense and correspond to a measure of relative classifiablity. As this is an important point and we also had a similar curiosity that drove your question and motivated in the first point, we did evaluate the CR-FIQA model as an FR embedding extraction model after every training epoch, and on 6 FR benchmarks (LFW, CPLFW, CALFW, Agedp-30, cfp-fp, cfp-ff). We attach here the log file of the training showing this performance at every epoch for the CR-FIQA(L) model. This log file is available in this GitHub repository when accessing pre-trained model (separate file "training.log"). One can notice that the performance is very comparable to the model trained under a similar training setting and only using the FR loss (here ArcFace, reported (https://arxiv.org/abs/1801.07698) e.g. on lfw ResNet100 trained on ArcFace loss reports an accuracy of 99.83% on LFW, 98.15% on agedb-30, and 92.08% on CPLFW, while the CR-FIQA(L) reaches 99.80% on LFW, 98.20% on Age-db30, and 92.90% on CPLFW). Thank you again for the interesting question and please do let us know if we can clarify this point further.

**Comment by Noobzm** [comment]
Got it, thanks again！
